data-processor/src/test/groovy/io/micronaut/data/processor/sql/ColumnTransformerSpec.groovy

```diff
@@ -111,7 +111,7 @@ class Project {
         ).query
 
         expect:
-        sql == 'UPDATE "project" SET "name"=UPPER(?),"org"=? WHERE (LOWER(name) = ? AND UPPER(org) = ?)'
+        sql == 'UPDATE "project" SET "name"=UPPER(?),"org"=? WHERE ("name" = UPPER(?) AND "org" = ?)'
```

Comment:
I would expect it to be `"name" = LOWER(?) AND "org" = UPPER(?)`

Reply:
No, the point is that name is written using `UPPER(?)` and org does not have write transform. So, in the database all names is written in UPPER and org as is. So when comparing given parameter in queries, we need to convert parameter for name to UPPER to match what is written in the db and for org don't need to convert. Both read transform will work, but for comparison it works correctly.

Reply:
Ah ok. Not sure if we can change this in v3.9. The only way would be to have it optional and make it the default in v4

Reply:
It should be backward compatible because we just change how WHERE clause works with data transform functions.